### PR TITLE
update Operator readme

### PR DESCRIPTION
**Pull request by taroface**
Fixes https://github.com/cockroachdb/cockroach-operator/issues/580.

@udnay @rail Should the content under `# Releases` be in `DEVELOPER.md` or a standalone file instead?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,81 +93,62 @@ Each pod should have `READY` status soon after being created.
...
-```
-kubectl apply -f example.yaml
-```
+> **Note:** You must scale by updating the `nodes` value in the Operator configuration. Using `kubectl scale statefulset <cluster-name> --replicas=4` will result in new pods immediately being terminated.
 
-## Stop the CockroachDB cluster
+### Upgrade the CockroachDB cluster
 
-Delete the StatefulSet:
+Perform a rolling upgrade by changing `image.name` in the custom resource. For details, see the [CockroachDB documentation](https://www.cockroachlabs.com/docs/v21.1/operate-cockroachdb-kubernetes#upgrade-the-cluster).
```
Comment (udnay) [inline README.md:144]: Do we want to use `stable` version here too?

**Review by udnay — APPROVED** [review]
A couple of questions but LGTM otherwise

**Comment by rail** [comment]
> Fixes #580.
> 
> @udnay @rail Should the content under `# Releases` be in `DEVELOPER.md` or a standalone file instead?

I think we can remove that section and point it to https://cockroachlabs.atlassian.net/wiki/spaces/KO/pages/1753841712/Cockroach+Operator+Release+Process

**Comment by taroface** [comment]
@rail That won't be visible to the public, will it? Or is this meant to be internal-only?

**Comment by taroface** [comment]
@udnay Ping for approval to merge.

**Review by udnay — APPROVED** [review]
LGTM

**Comment by rail** [comment]
> @rail That won't be visible to the public, will it? Or is this meant to be internal-only?

There is nothing secret in that doc, but I"m not sure if it useful for others, because it describes our work flow.

**Pull request merged**